src/test/java/io/vavr/control/TryTest.java

```diff
@@ -969,6 +969,15 @@ public void shouldConvertFailureToEitherLeftSupplier() {
         assertThat(failure().toEither(() -> "test").isLeft()).isTrue();
     }
 
+    @Test
+    public void shouldConvertFailureToEitherLeftMapper() {
+        final Try<Object> failure = Try.failure(new RuntimeException("a certain value"));
+        final Either<Boolean, Object> either = failure.toEither(t ->
+                t.getMessage().equals("a certain value")
+        );
+        assertThat(either).isEqualTo(Either.left(true));
```

Comment:
I'd probably replace the test code in a way that limits the number of comparisons to one, which would be the one in the assertion itself.
You could use a mapper function that extracts the message from the exception and use that as the left value to compare to in the assertion. This would make the code both shorter, and — in my opinion — easier to understand, since it would have a slightly less degree of indirection.
```suggestion
        final Either<Boolean, Object> either = failure.toEither(Throwable::getMessage)
        assertThat(either).isEqualTo(Either.left("a certain value"));
```

Reply:
Sure, sounds much better that way 👍 I also felt a bit weird using `Throwable => Boolean` as my function where many people would maybe expect that to then be `Predicate<Throwable>`. I can see also how the `equals` looks like an assertion.

Main point was just to have a clear expected value returned by the function, so the suggestion sounds great